Std/Tactic/Exact.lean

```diff
@@ -0,0 +1,19 @@
+/-
+Copyright (c) 2023 Mario Carneiro. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro
+-/
+import Lean
+
+/-! # `exact` tactic (`MetaM` version) -/
+
+open Lean Meta
+
+/--
+`MetaM` version of `Lean.Elab.Tactic.evalExact`: add `mvarId := x` to the metavariable assignment.
+This method wraps `Lean.MVarId.assign`, checking whether `mvarId` is already assigned, and whether
+the expression has the right type. -/
+def Lean.MVarId.myExact (g : MVarId) (e : Expr) : MetaM Unit := do
```

Comment:
`myExact` sounds more like a tutorial tactic to me that we would e.g. write in the meta programming book as a show case, is there a reason you didn't go for just `exact`?

Reply:
And indeed the name is a placeholder, Mario wants to change it but didn't have an idea yet.

Reply:
How about `Lean.MVarId.assign'` so that it pops up in the autocomplete for `assign`? Or `Lean.MVarId.assignIfDefeq`?

By the way, the first line of the definition could be `guard <| ← isDefEq (← g.getType) (← inferType e)`.

Reply:
```suggestion
def Lean.MVarId.assignIfDefeq (g : MVarId) (e : Expr) : MetaM Unit := do
```